### PR TITLE
[[ Bug 14413 ]] revZip: add UTF-8 understanding of archive filename / archive item names on Windows

### PR DESCRIPTION
**Pull request by livecodesebastien**
This pull request only contains the bugfix note.

The bugfix is on the pull request https://github.com/runrev/livecode-thirdparty/pull/23.

### CODE REVIEW

**Comment by runrevmark** [comment]
What aspects of 'UTF-8' support does this address on Windows?

**Comment by livecodesebastien** [comment]
I renamed the pull request, to give a more precise description.

**Comment by runrevmark** [comment]
@runrevsebastien: I think you misunderstood what I meant here... You've added the conversion functions to external.c - they should be V0 external API calls... So the _engine_ should do the work. The reason to do it this way is that we may very well want to make the 'native text encoding' configurable to allow subtle variations to be accounted for in script.

Also, I believe (but may be wrong) we used to read/write native filenames into zip archives on Mac and Linux - i.e. MacRoman / 8859-1.

File: revzip/src/revzip.cpp [hunk 1]
```diff
@@ -1267,15 +1267,47 @@ void revZipEnumerateItems(char *p_arguments[], int p_argument_count, char **r_re
...
+			}
+			else
+			{
+				// SN-2015-03-10: [[ Bug 14413 ]] We convert the string to UTF-8
+				//  in case it was natively encoded, as revZipEnumerateItems is
+				//  meant to return a UTF-8 encoded string.
+				const char *t_converted_name;
+                int t_success;
+
+				if (t_stat.bitflags && ZIP_UTF8_FLAG)
```
Comment (runrevmark) [inline revzip/src/revzip.cpp]: This should be '&' - bitwise and.

**Comment by runrevmark** [comment]
In light of Bug 9185 (http://quality.runrev.com/show_bug.cgi?id=9185) I wonder if we have gone far enough here. Indeed, we should perhaps expose a 'ConvertPathFromLiveCodeToPlatform' function - this would take a filename as a UTF-8 string (in LiveCode 'format') and return the platform-specific (absolute) version. You'll need to check with @livecodefraser here, but on Windows the only way to get >259 chars in a filename is to either short-filepath it or use the extended 'Unicode' filename variant (i.e. I suspect just using the W-char version of fopen won't do the trick).

**Comment by runrevmark** [comment]
Ignore my previous comment - that's a distinct issue.

Reviewed-by: @runrevmark.

**Pull request merged**